### ISSUE TITLE
Some lexers lack aliases

### ISSUE THREAD

**Issue by 8day** [pygments/pygments#2630]
As far as I can tell, `get_lexer_by_name()` is in fact `get_lexer_by_alias()`. So if lexer name is not listed under aliases, you won't be able to find such lexer through `get_lexer_by_name()`.

ATM at least `lexers/special.py:TextLexer` and `lexers/special.py:OutputLexer` don't have their names present in their aliases.
```python
class TextLexer(Lexer):
    """
    "Null" lexer, doesn't highlight anything.
    """
    name = 'Text only'
    aliases = ['text'] # This should be ['text only', 'text'].
    filenames = ['*.txt']
    mimetypes = ['text/plain']
    priority = 0.01

    def get_tokens_unprocessed(self, text):
        yield 0, Text, text

    def analyse_text(text):
        return TextLexer.priority
```

**Comment by jeanas** [comment]
Yes, the name `get_lexer_by_name()` is unfortunately misleading. However, it's not the intent that `lexer.name` should always be part of `lexer.aliases`. Basically, the name is meant to be a human-readable language name (esp. for listing at https://pygments.org/languages), and `lexer.aliases` is for lookup, e.g., in reST/Markdown code blocks.

**Comment by birkenfeld** [comment]
Yeah, maybe the docs need a tweak.
